### PR TITLE
Charge redistribution

### PR DESCRIPTION
**Pull request by rmatsum836**
## Description
Addressing issue discussed in #13.  I added the argument `charge_tol` to `ante_charges`.  If the absolute value of the `net_charge` minus the charges assigned by antechamber is greater than `charge_tol`, then an error will be raised.  If the absolute value is less than `charge_tol` but greater than zero, the net charge will be evenly distributed across all atoms of the molecule.

Currently `charge_tol` is set to 5e-3, but I would be open to changing this.  I also added tests to check that 1) charges get redistributed correctly and 2) an error gets raised if abs(net_charge - antechamber charges) > `charge_tol`.

I also ran black on the files I edited which is why a lot of lines got reformatted.

## Questions
- [ ] Question1

## Status
- [x] Ready to go

### CODE REVIEW

File: antefoyer/antefoyer.py [hunk 1]
```diff
@@ -121,46 +129,69 @@ def ante_charges(molecule, charge_style, net_charge=0.0,
...
-                _antechamber_error(out,err,workdir)
+            if "Fatal Error" in err or proc.returncode != 0:
+                _antechamber_error(out, err, workdir)
 
             # Now read in the mol2 file with atomtyping
-            charges = pmd.load_file('ante_out.mol2',structure=True)
+            charges = pmd.load_file("ante_out.mol2", structure=True)
+
+            charge_list = [atom.charge for atom in charges]
+            if (net_charge - sum(charge_list)) > charge_tol:
```
Comment (rsdefever) [inline antefoyer/antefoyer.py]: Do we need absolute values here to handle positive or negative net charges?

File: antefoyer/antefoyer.py [hunk 2]
```diff
@@ -121,46 +129,69 @@ def ante_charges(molecule, charge_style, net_charge=0.0,
...
+                raise ValueError(
+                    "The sum of charges defined by antechamber"
+                    " is {}, which differs from the desired net charge"
+                    " of {} by a value greater than {}".format(
+                        sum(charge_list), net_charge, charge_tol
+                    )
+                )
+            elif (net_charge - sum(charge_list)) < charge_tol:
```
Comment (rsdefever) [inline antefoyer/antefoyer.py]: Same here.

**Review by rsdefever — CHANGES_REQUESTED** [review]
I haven't downloaded and run tests yet, but here are my initial comments.

File: antefoyer/antefoyer.py (hunk 1 above)
Comment (rmatsum836) [reply antefoyer/antefoyer.py]: Oops, forgot to push the absolute value.

File: antefoyer/tests/test_antefoyer.py [hunk 3]
```diff
@@ -17,65 +18,88 @@
...
+                charges = ante_charges(ethane, "bcc", net_charge=-1)
+            assert isfile("ante_errorlog.txt")
+
+
+@pytest.mark.skipif(ANTECHAMBER is None, reason="antechamber is not installed")
+def test_ante_charge_delta():
+    ethane = pmd.load_file(get_fn("ethane.mol2"), structure=True)
+    charges = ante_charges(ethane, "bcc", net_charge=0)
 
+    assert np.allclose(sum([i.charge for i in charges]), 0, atol=5e-3)
```
Comment (rsdefever) [inline antefoyer/tests/test_antefoyer.py]: Why do we need the `atol` here? These charges should be adjusted, right? So they should be spot-on.

File: antefoyer/tests/test_antefoyer.py [hunk 4]
```diff
@@ -17,65 +18,88 @@
...
+
+
+@pytest.mark.skipif(ANTECHAMBER is None, reason="antechamber is not installed")
+def test_charge_tolerance():
+    with pytest.raises(ValueError, match=r"The sum of charges"):
+        ethane = pmd.load_file(get_fn("ethane.mol2"), structure=True)
+        ante_charges(ethane, "bcc", charge_tol=0.001)
```
Comment (rsdefever) [inline antefoyer/tests/test_antefoyer.py:105]: Curiosity? How far off are the net charges on Ethane?

**Review by rsdefever — COMMENTED** [review]
Couple more small questions, then LGTM.

File: antefoyer/tests/test_antefoyer.py (hunk 3 above)
Comment (rmatsum836) [reply antefoyer/tests/test_antefoyer.py]: This doesn't make sense revisiting this morning.  I am removing the `atol` argument.

File: antefoyer/tests/test_antefoyer.py (hunk 4 above)
Comment (rmatsum836) [reply antefoyer/tests/test_antefoyer.py:105]: Ethane is off by `0.0020000000000000087`.

**Pull request merged**